addons/mail/models/res_users.py

```diff
@@ -144,7 +145,17 @@ def write(self, vals):
                 )
         if 'notification_type' in vals:
             for user in user_notification_type_modified:
-                self.env["bus.bus"]._sendone(user.partner_id, "mail.record/insert", {"Persona": {"id": user.partner_id.id, "type": "partner", "notification_preference": user.notification_type}})
+                self.env["bus.bus"]._sendone(
+                    user.partner_id,
+                    "mail.record/insert",
+                    Store(
+                        "res.partner",
```

Comment:
:heart:

Reply:
Hello, here's a little bit more details, as you seem to be interested in this particular line, and the PR message might be lacking some information about future steps.

I'm guessing you have concerns about formatting and verbosity, and I agree that this intermediate step/PR makes the code a little more verbose in some places, while others are already simplified.
Places like this one were many lines appear are highlighting a deeper issue on how hard it is currently for us to send data.
Hopefully I can reassure you with sharing some thoughts on what's coming.

This can eventually become: `Store(user, fields=["notification_type"]).get_result(),`

And combined with https://github.com/odoo/odoo/pull/161495, and with an extra method for store, I'm expecting 99% of our bus send to look like
`user._bus_send_store(Store(user, fields=["notification_type"]))`
Or even
`user._bus_send_store(user, fields=["notification_type"])`
Depending on how far we want to simplify, and the syntax we choose to keep.

I hope we can all agree this will be much better than the completely bloated starting point of
`self.env["bus.bus"]._sendone(user.partner_id, "mail.record/insert", {"Persona": {"id": user.partner_id.id, "type": "partner", "notification_preference": user.notification_type}})`

This is planned for future steps, that are coming in the next days or weeks, but this particular PR was getting pretty big already so I prefer to move forward step by step. It was important to standardize the various calls here before going further.

Reply:
Actually it was more about removing "Persona" and use "res.partner" :disguised_face:

Reply:
> Actually the ❤️ was more about removing "Persona" and use "res.partner" 🥸

Note that "persona" concepts is still necessary as an abstraction for `res.partner` or `mail.guest`. We have it in place to reduce LOCs with branching. Here it works fine to replace the `Persona` by `res.partner` because a `res.partner` is necessarily a `Persona`.

I get this "Persona" concepts is not great, especially when this doesn't map to a genuine python model... We also agree with this, but this is made to solve a practical issue hence why this is still there.

Probably in the future this can be improved to feel more natural, e.g. with a genuine python model or better JS code where partner and guests work more seamlessly without explicitly defining a "Persona" abstraction... But this is not necessarily doable with the added cost this could imply (more smart JS code and/or more complex python modelling), and since we want at least a solution that's why we went with "Persona" abstraction.

Reply:
Yep I know, and indeed it does not make much sense outside of discuss currently. Even modules like project sharing, knowledge or "upcoming google drike like" documents won't let anonymous people deal with update, it will be globally a readonly access so we don't really need a lot of machinery. Some tokens and dedicated routes are sufficient for flows that require it.

There is some pâté to clean with res.partner, mail.guest, website.visitor, res.users (you may add hr.employee on top of it) ... especially we want to support mailing features without creating partners (which means supporting emails only someday). But we already tried various stuff with Antony and in the end, abstractions or trying to use mixins add complexity and code for few readability. A few if-else are maybe more low-tech but understandable ^^ .

Anyway voila.